pkg/comp-functions/functions/common/rbac.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"context"
+
+	xkubev1 "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddSaWithRole creates a service account with the given policy and binds it to the role.
+func AddSaWithRole(ctx context.Context, iof *runtime.Runtime, policies []rbacv1.PolicyRule, compName, namespace, suffix string) error {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sa-" + suffix,
+			Namespace: namespace,
+		},
+	}
+
+	err := iof.Desired.PutIntoObject(ctx, sa, compName+"-"+suffix+"-serviceaccount")
+	if err != nil {
+		return err
+	}
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      suffix + "-role",
+			Namespace: namespace,
+		},
+		Rules: policies,
+	}
+
+	saReference := xkubev1.Reference{
+		DependsOn: &xkubev1.DependsOn{
+			Name: compName + "-" + suffix + "-serviceaccount",
+		},
+	}
```

Comment:
I can see potential here for exporting this string to single variable

Reply:
that one: `compName+"-"+suffix+"-role"`

Reply:
I'm not sure I understand what you mean here?

Reply:
in line 37 and 22 You use the same multiple variables chains, I'd just replace it with serviceAccountName := fmt.Sprintf("%s-%s-%s-serviceaccount")

Reply:
Ah I see, yes I can optimize this. Also if other chains repeat as well.